madara/crates/primitives/transactions/src/from_blockifier.rs

```diff
@@ -5,47 +5,57 @@ use blockifier::transaction::transaction_execution::Transaction as BTransaction;
 impl From<BTransaction> for TransactionWithHash {
     fn from(tx: BTransaction) -> TransactionWithHash {
         match tx {
-            BTransaction::AccountTransaction(tx) => tx.into(),
-            BTransaction::L1HandlerTransaction(tx) => tx.into(),
+            BTransaction::Account(tx) => tx.tx.into(),
+            BTransaction::L1Handler(tx) => tx.into(),
         }
     }
 }
 
-use blockifier::transaction::account_transaction::AccountTransaction as BAccountTransaction;
-impl From<BAccountTransaction> for TransactionWithHash {
-    fn from(tx: BAccountTransaction) -> TransactionWithHash {
+use starknet_api::executable_transaction::Transaction as ApiTransaction;
+impl From<ApiTransaction> for TransactionWithHash {
+    fn from(tx: ApiTransaction) -> TransactionWithHash {
         match tx {
-            BAccountTransaction::Declare(tx) => tx.into(),
-            BAccountTransaction::DeployAccount(tx) => tx.into(),
-            BAccountTransaction::Invoke(tx) => tx.into(),
+            ApiTransaction::Account(tx) => tx.into(),
+            ApiTransaction::L1Handler(tx) => tx.into(),
         }
     }
 }
 
-use blockifier::transaction::transactions::DeclareTransaction as BDeclareTransaction;
-impl From<BDeclareTransaction> for TransactionWithHash {
-    fn from(tx: BDeclareTransaction) -> TransactionWithHash {
+use starknet_api::executable_transaction::AccountTransaction as ApiAccountTransaction;
+impl From<ApiAccountTransaction> for TransactionWithHash {
+    fn from(tx: ApiAccountTransaction) -> TransactionWithHash {
+        match tx {
+            ApiAccountTransaction::Declare(tx) => tx.into(),
+            ApiAccountTransaction::DeployAccount(tx) => tx.into(),
+            ApiAccountTransaction::Invoke(tx) => tx.into(),
+        }
```

Comment:
Is it possible to collapse this to
```suggestion
        match tx {
            ApiAccountTransaction::Declare(tx)
            | ApiAccountTransaction::DeployAccount(tx)
            | ApiAccountTransaction::Invoke(tx) => tx.into(),
        }
```
Or is each `tx` a different type?

Reply:
they are not the same internal types